ports/stm32/Makefile

```diff
@@ -93,7 +93,7 @@ CFLAGS_MCU_l4 = $(CFLAGS_CORTEX_M) -mtune=cortex-m4 -mcpu=cortex-m4
 CFLAGS_MCU_h7 = $(CFLAGS_CORTEX_M) -mtune=cortex-m7 -mcpu=cortex-m7
 CFLAGS_MCU_wb = $(CFLAGS_CORTEX_M) -mtune=cortex-m4 -mcpu=cortex-m4
 
-CFLAGS += $(INC) -Wall -Wpointer-arith -Werror -std=gnu99 -nostdlib $(CFLAGS_MOD) $(CFLAGS_EXTRA)
+CFLAGS += $(INC) -Wall -Wpointer-arith -Wdouble-promotion -Wfloat-conversion -std=gnu99 -nostdlib $(CFLAGS_MOD) $(CFLAGS_EXTRA)
```

Comment:
As for esp32, `-Werror` will need to stay.  To suppress the resulting warnings/errors might be easy enough to add something like this:
```
$(BUILD)/$(HAL)/Src/%.o: CFLAGS += -Wno-double-promotion
```

Reply:
@dpgeorge  Sorry I'm currently in the process to figure out what I can enable and what not, hence the temporary removal of `-Werror`: no time for review yet, should have done this in a separate branch :)

Reply:
Ahh, no worries!  Just make a note when it's done.

Reply:
Does it help if I click 'Resolve conversation' to indicate an item I've worked on? It's abit confusing because in Gitlab 'Resove conversation' is for the reviewer to indicate something is done, here it's the other way around?

Reply:
Yes, you can "resolve conversation" to indicate you dealt with it.  But it's not critical, I will still check all points on final review.  Just make a global comment when this PR is ready to review again.